broker/lib/controllers/ServiceFabrikApiController.js

```diff
@@ -32,6 +35,72 @@ const CloudControllerError = {
   }
 };
 
+
+function getResourceAnnotationStatus(resourceType, resourceId, guid, start_state, started_at) {
+  logger.info(`getResourceAnnotationStatus is called`);
+  return Promise.delay(CONST.EVENTMESH_POLLER_DELAY)
+    .then(() => eventmesh.server.getAnnotationState({
+      resourceType: resourceType,
+      resourceId: resourceId,
+      annotationName: 'backup',
+      annotationType: 'default',
+      annotationId: guid
+    })).then(state => {
+      const duration = (new Date() - started_at) / 1000;
+      const backup_start_timeout = 15 //sec
+      logger.info(`Lock duration : ${duration} `);
+      if (duration > backup_start_timeout) {
+        throw new Timeout(`Backup not picked up from queue`);
+      }
+      if (state === start_state) {
+        logger.info('Waiting to get the annotation state');
+        return getResourceAnnotationStatus(resourceType, resourceId, guid, start_state, started_at);
+      } else if (state === 'error') {
+        return eventmesh.server.getAnnotationKeyValue({
+            resourceType: resourceType,
+            resourceId: resourceId,
+            annotationName: 'backup',
+            annotationType: 'default',
+            annotationId: guid,
+            key: 'result'
+          })
+          .then(error => {
+            logger.info('Operation reported error', JSON.parse(error));
+            let json = JSON.parse(error)
+            let message = json.message;
+            if (json.error && json.error.description) {
+              message = `${message}. ${json.error.description}`
+            }
+            let err;
+            switch (json.status) {
```

Comment:
we already have HTTP status codes in constants.js - https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/common/constants.js#L55-L67

Reply:
Used constants.